server/src/main/java/com/cloud/network/Ipv6ServiceImpl.java

```diff
@@ -0,0 +1,523 @@
+
+    @Inject
+    NetworkOfferingDao networkOfferingDao;
+    @Inject
+    VlanDao vlanDao;
+    @Inject
+    DataCenterGuestIpv6PrefixDao dataCenterGuestIpv6PrefixDao;
+    @Inject
+    Ipv6GuestPrefixSubnetNetworkMapDao ipv6GuestPrefixSubnetNetworkMapDao;
+    @Inject
+    FirewallRulesDao firewallDao;
+    @Inject
+    FirewallService firewallService;
+    @Inject
+    NetworkDetailsDao networkDetailsDao;
+    @Inject
+    NicDao nicDao;
+    @Inject
+    DomainRouterDao domainRouterDao;
+    @Inject
+    AccountManager accountManager;
+    @Inject
+    NetworkModel networkModel;
+    @Inject
+    FirewallManager firewallManager;
+
+    protected void releaseIpv6Subnet(long subnetId) {
+        Ipv6GuestPrefixSubnetNetworkMapVO ipv6GuestPrefixSubnetNetworkMapVO = ipv6GuestPrefixSubnetNetworkMapDao.createForUpdate(subnetId);
+        ipv6GuestPrefixSubnetNetworkMapVO.setState(Ipv6GuestPrefixSubnetNetworkMap.State.Free);
+        ipv6GuestPrefixSubnetNetworkMapVO.setNetworkId(null);
+        ipv6GuestPrefixSubnetNetworkMapVO.setUpdated(new Date());
+        ipv6GuestPrefixSubnetNetworkMapDao.update(ipv6GuestPrefixSubnetNetworkMapVO.getId(), ipv6GuestPrefixSubnetNetworkMapVO);
+    }
+
+    @Override
+    public boolean start() {
+        _ipv6GuestPrefixSubnetNetworkMapStateScanner.scheduleWithFixedDelay(new Ipv6GuestPrefixSubnetNetworkMapStateScanner(), 300, 30*60, TimeUnit.SECONDS);
+        return true;
+    }
+
+    @Override
+    public boolean configure(String name, Map<String, Object> params) throws ConfigurationException {
+        _name = name;
+        _configParams = params;
+        _ipv6GuestPrefixSubnetNetworkMapStateScanner = Executors.newScheduledThreadPool(1, new NamedThreadFactory("Ipv6GuestPrefixSubnet-State-Scanner"));
+
+        return true;
+    }
+
+    @Override
+    public List<Class<?>> getCommands() {
+        final List<Class<?>> cmdList = new ArrayList<Class<?>>();
+        cmdList.add(CreateIpv6FirewallRuleCmd.class);
+        cmdList.add(ListIpv6FirewallRulesCmd.class);
+        cmdList.add(UpdateIpv6FirewallRuleCmd.class);
+        cmdList.add(DeleteIpv6FirewallRuleCmd.class);
+        return cmdList;
+    }
+
+    @Override
+    public String getConfigComponentName() {
+        return Ipv6Service.class.getSimpleName();
+    }
+
+    @Override
+    public ConfigKey<?>[] getConfigKeys() {
+        return new ConfigKey<?>[] {
+                Ipv6NetworkOfferingCreationEnabled
+        };
+    }
+
+    @Override
+    public Pair<Integer, Integer> getUsedTotalIpv6SubnetForPrefix(DataCenterGuestIpv6Prefix prefix) {
+        List<Ipv6GuestPrefixSubnetNetworkMapVO> usedSubnets = ipv6GuestPrefixSubnetNetworkMapDao.listUsedByPrefix(prefix.getId());
+        final IPv6Network ip6Prefix = IPv6Network.fromString(prefix.getPrefix());
+        Iterator<IPv6Network> splits = ip6Prefix.split(IPv6NetworkMask.fromPrefixLength(IPV6_SLAAC_CIDR_NETMASK));
+        int total = 0;
```

Comment:
@shwstppr 
I have tested with `2022:2:2:2::/60`, it should have 16 subnets but this returns 15.

Reply:
@weizouapache this is my mistake. I was getting confused with the first subnet (2022:2:2:2::/64 in your prefix of 2022:2:2:2::/60) so I added code for skipping it. Will fix it